### PR TITLE
[HttpFoundation] fixed bag registration after session start

### PR DESCRIPTION
**Pull request by digilist**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #10707 |
| License | MIT |
| Doc PR |  |

I suggested this fix some time ago, but there were no resoponses. Hopefully, this PR finds some attention :smiley:

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Tests/Session/Storage/NativeSessionStorageTest.php [hunk 1]
```diff
@@ -254,4 +254,19 @@ public function testCanStartOutside54()
         $this->assertFalse(isset($_SESSION[$key]));
         $storage->start();
     }
+
+    public function testRegisterBagAfterStarting()
+    {
+        $storage = $this->getStorage();
+        $storage->start();
```
Comment (iltar) [inline src/Symfony/Component/HttpFoundation/Tests/Session/Storage/NativeSessionStorageTest.php]: Can you also add a test to see what happens when you register before starting?

Comment (digilist) [inline src/Symfony/Component/HttpFoundation/Tests/Session/Storage/NativeSessionStorageTest.php]: Yes, see above. Did you expect it this way?

File: src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php [hunk 2]
```diff
@@ -422,12 +426,28 @@ protected function loadSession(array &$session = null)
...
     }
+
+    /**
+     * Load and initialize the session for the given bag and pass the
+     * session variable into the bag.
+     *
+     * @param SessionBagInterface $bag
+     * @param array|null          $session
+     */
+    protected function loadSessionBag(SessionBagInterface $bag, array &$session = null)
```
Comment (jakzal) [inline src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php]: Should be private.

**Comment by fabpot** [comment]
Closing as the "right" fix would instead be to throw an exception when trying to register a bag after the session is started.

**Pull request closed without merging**